### PR TITLE
ODLC state unit test

### LINKED ISSUE #144: ODLC state unit test
# ODLC Unit Test

## Problem

Once the ODLC state is implemented, we want a file to test the functionality of said state.

## Solution

Make a unit test that tests all the functionalities of the ODLC state.

## Additional Information

Provide any additional information.

### PR DESCRIPTION
**Pull request by Ooglely**
A unit test for the ODLC state so that we can test its functionality as a standalone state.

### CODE REVIEW

**Comment by Ooglely** [comment]
Waiting on the ability to skip states in the state machine before ready for review

File: flight/test_files/odlc_state_test.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+from state_machine.flight_settings import FlightSettings
+
+
+def start_state_machine(state_machine: StateMachine) -> None:
+    """
+    add
+    """
+    asyncio.run(state_machine.run())
+
+
```
Comment (Flamingpuffins) [inline flight/test_files/odlc_state_test.py:25]: add docstring

File: state_machine/states/impl/waypoint_impl.py [hunk 2]
```diff
@@ -102,7 +105,8 @@ async def run(self: Waypoint) -> State:
             )
 
             curr_altitude: float = drone_position.relative_altitude_m
-            altitude_slope: float = (waypoint.altitude - curr_altitude) / path_length
+            # altitude_slope: float = (waypoint.altitude - curr_altitude) / path_length
+            # removing this because "too many local variables" eww
```
Comment (cjhr95) [inline state_machine/states/impl/waypoint_impl.py]: Change or remove this comment for professionalism's sake.

File: flight/test_files/odlc_state_test.py [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+    flight_settings = FlightSettings(sim_flag=_sim, skip_waypoint=True)
+    state_machine = StateMachine(Start(drone, flight_settings), drone, flight_settings)
+    state_machine_process: Process = Process(
+        target=start_state_machine,
+        args=(state_machine,),
+    )
+    state_machine_process.start()
+
+    activated_odlcs = 0
+    while activated_odlcs != 5:
```
Comment (cjhr95) [inline flight/test_files/odlc_state_test.py]: Change this to a variable or a constant; hard-coded values are usually a no-go

**Review by cjhr95 — CHANGES_REQUESTED** [review]

File: flight/test_files/odlc_state_test.py [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+
+    Parameters
+    ----------
+    _sim: bool
+        Whether or not the test is being run in a simulation
+    """
+    logging.basicConfig(level=logging.INFO)
+    drone = Drone()
+    flight_settings = FlightSettings(sim_flag=_sim, skip_waypoint=True)
+    state_machine = StateMachine(Start(drone, flight_settings), drone, flight_settings)
```
Comment (cjhr95) [inline flight/test_files/odlc_state_test.py]: type hints here

File: flight/test_files/odlc_state_test.py [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+    state_machine_process: Process = Process(
+        target=start_state_machine,
+        args=(state_machine,),
+    )
+    state_machine_process.start()
+
+    activated_odlcs = 0
```
Comment (cjhr95) [inline flight/test_files/odlc_state_test.py]: Needs type hint

File: flight/test_files/odlc_state_test.py [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+    while activated_odlcs != odlc_count:
+        try:
+            with open("flight/data/output.json", "r", encoding="UTF-8") as file:
+                output_data = json.load(file)
```
Comment (cjhr95) [inline flight/test_files/odlc_state_test.py]: Need a type hint here as well

**Review by cjhr95 — CHANGES_REQUESTED** [review]
Missing a few, quick type hints

**Review by cjhr95 — APPROVED** [review]
Nice job kiddos

**Pull request merged**